### PR TITLE
build: skip det-deploy-local for webui only changes

### PR DESCRIPTION
**Pull request by hamidzr**
## Commentary (optional)

e2e-test is usually the longest running workflow. cut it from 35min to 8min.
the idea is that the skipped tests do not test changes introduced in /webui

more context: https://determined-ai.slack.com/archives/CT01QS70Q/p1653505930158909

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by NicholasBlaskey — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -1031,6 +1031,7 @@ jobs:
     steps:
       - checkout
       - skip-if-docs-only
+      - skip-if-webui-only
```
Comment (hkang1) [inline .circleci/config.yml:1034]: **super nit:** `skip-if-web-only` instead of `skip-if-webui-only`? or maybe @trentwatt 's PR already covers this

**Review by hkang1 — APPROVED** [review]
Smart change!

File: .circleci/config.yml (hunk 1 above)
Comment (hamidzr) [reply .circleci/config.yml:1034]: hmm, IMO it's best to wait for the pr that's updating these to do it. as is most of the references are `webui` 

(21 occurrences here alone)

**Comment by hamidzr** [comment]
https://app.circleci.com/pipelines/github/determined-ai/determined/24977/workflows/a9b7bfc4-ccc2-49f6-8e5e-2a812b54f62d test

**Pull request merged**